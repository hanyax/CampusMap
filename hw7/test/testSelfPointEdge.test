# This test test if the edge can have a the same parentNode and childrenNode

# Create a graph
CreateGraph graph1

# Add nodes
AddNode graph1 n1
AddNode graph1 n2

# Add an edge
AddEdge graph1 n1 n1 0.0
AddEdge graph1 n1 n1 0.7
AddEdge graph1 n2 n2 0.0
AddEdge graph1 n2 n2 0.5

# Print the nodes in the graph and the outgoing edges from n1
ListNodes graph1
ListChildren graph1 n1
ListChildren graph1 n2

#Test Find Path SelfPoint
FindPath graph1 n1 n1
