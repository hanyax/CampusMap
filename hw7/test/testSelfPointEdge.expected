# This test test if the edge can have a the same parentNode and childrenNode

# Create a graph
created graph graph1

# Add nodes
added node n1 to graph1
added node n2 to graph1

# Add an edge
added edge 0.000 from n1 to n1 in graph1
added edge 0.700 from n1 to n1 in graph1
added edge 0.000 from n2 to n2 in graph1
added edge 0.500 from n2 to n2 in graph1

# Print the nodes in the graph and the outgoing edges from n1
graph1 contains: n1 n2
the children of n1 in graph1 are: n1(0.000) n1(0.700)
the children of n2 in graph1 are: n2(0.000) n2(0.500)

#Test Find Path SelfPoint
path from n1 to n1:
total cost: 0.000
